bec_widgets/utils/bec_widget.py

```
@@ -218,6 +218,44 @@ def screenshot(self, file_name: str | None = None):
         screenshot.save(file_name)
         logger.info(f"Screenshot saved to {file_name}")
 
+    def screenshot_bytes(
+        self,
+        *,
+        max_width: int | None = None,
+        max_height: int | None = None,
+        fmt: str = "PNG",
+        quality: int = -1,
+    ) -> QByteArray:
+        """
+        Grab this widget, optionally scale to a max size, and return encoded image bytes.
+
+        If max_width/max_height are omitted (the default), capture at full resolution.
+
+        Args:
+            max_width(int, optional): Maximum width of the screenshot.
+            max_height(int, optional): Maximum height of the screenshot.
+            fmt(str, optional): Image format (e.g., "PNG", "JPEG").
+            quality(int, optional): Image quality (0-100), -1 for default.
+
+        Returns:
+            QByteArray: The screenshot image bytes.
+        """
+        if not isinstance(self, QWidget):
+            return QByteArray()
+        pixmap: QPixmap = self.grab()
```

maybe raise here if grab is not defined. Might be easier to debug